server/src/http/jwt.rs

```diff
@@ -0,0 +1,114 @@
+use crate::http::claims::JwtClaims;
+use crate::http::state::AppState;
+use axum::{
+    extract::State,
+    http::{Request, StatusCode},
+    middleware::Next,
+    response::Response,
+};
+use iggy::error::Error;
+use iggy::utils::timestamp::TimeStamp;
+use jsonwebtoken::{encode, Algorithm, DecodingKey, EncodingKey, Header, TokenData, Validation};
+use std::sync::Arc;
+
+const AUDIENCE: &str = "iggy.rs";
+const AUTHORIZATION: &str = "authorization";
+const BEARER: &str = "Bearer ";
+
+#[derive(Debug, Clone)]
+pub struct Identity {
+    pub user_id: u32,
+}
+
+pub async fn jwt_auth<T>(
+    State(state): State<Arc<AppState>>,
+    mut request: Request<T>,
+    next: Next<T>,
+) -> Result<Response, StatusCode> {
+    match request.uri().path() {
+        "/" => return Ok(next.run(request).await),
+        "/ping" => return Ok(next.run(request).await),
+        "/users/login" => return Ok(next.run(request).await),
+        _ => (),
+    }
```

Comment:
if you define these on top:
```rust
const UNAUTHORIZED: StatusCode = StatusCode::UNAUTHORIZED;
const AUTHORIZATION: &str = "Authorization";
const BEARER: &str = "Bearer ";
```

you can use them later as:
```rust
    let header = request
        .headers()
        .get(AUTHORIZATION)
        .ok_or(UNAUTHORIZED)?;
```
Looks a bit cleaner.